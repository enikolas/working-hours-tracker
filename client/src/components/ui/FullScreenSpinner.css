.full-screen-spinner-layer {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 100;
  background: rgba(37,67,86,0.5);
}
