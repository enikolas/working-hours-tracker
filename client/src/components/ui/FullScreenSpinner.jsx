import React from 'react';
import PropTypes from 'prop-types';

import Spinner from '../ui/Spinner';

import './FullScreenSpinner.css';

const FullScreenSpinner = props => (
  props.active ?
    (
      <div className="full-screen-spinner-layer">
        <Spinner
          class="loadingPage"
        />
      </div>
    )
    : ''
);

FullScreenSpinner.propTypes = {
  active: PropTypes.bool
};

FullScreenSpinner.defaultProps = {
  active: false
};

export default FullScreenSpinner;
