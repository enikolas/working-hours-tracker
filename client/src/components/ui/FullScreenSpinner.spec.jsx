import React from 'react';
import { shallow } from 'enzyme';

import FullScreenSpinner from './FullScreenSpinner';

describe('FullScreenSpinner', () => {
	it('should render properly withouth props', () => {
		const wrapper = shallow(<FullScreenSpinner />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should render properly with active', () => {
		const wrapper = shallow(<FullScreenSpinner active />);
		expect(wrapper).toMatchSnapshot();
	});
});
