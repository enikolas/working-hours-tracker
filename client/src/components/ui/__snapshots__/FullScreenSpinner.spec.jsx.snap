// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenSpinner should render properly with active 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FullScreenSpinner
    active={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Spinner
        class="loadingPage"
      />,
      "className": "full-screen-spinner-layer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "class": "loadingPage",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Spinner
          class="loadingPage"
        />,
        "className": "full-screen-spinner-layer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "class": "loadingPage",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`FullScreenSpinner should render properly withouth props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FullScreenSpinner
    active={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "",
  Symbol(enzyme.__nodes__): Array [
    "",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
